Fix SBS dialog (Addressed Review)

jroweboy opened the pull request:
Picking up where https://github.com/citra-emu/citra/pull/4605 left off. See that for more context. This is just me trying to get it merged since its holding up other PRs and really should be merged.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/citra-emu/citra/4757)
<!-- Reviewable:end -->

B3n30 commented on src/citra_qt/util/sbs_dialog.cpp:
```diff
@@ -0,0 +1,82 @@
+
+#include <algorithm>
+#include <QMessageBox>
+#include <QWidget>
```
Both includes are still unnecessary since they are in the header

B3n30 commented on src/citra_qt/util/sbs_dialog.cpp:
```diff
@@ -0,0 +1,82 @@
+#include "citra_qt/util/sbs_dialog.h"
+#include "core/settings.h"
+
+// move dialog to centre of left eye in SBS 3D mode
```
Should be in header

B3n30 commented on src/citra_qt/util/sbs_dialog.cpp:
```diff
@@ -0,0 +1,82 @@
+                 screen_geometry.center().y() - dialog_height / 2);
+}
+
+// move dialog to centre of right eye in SBS 3D mode
```
Dito

jroweboy commented:
@zhaowenlan1779 please look it over and merge it when you are done

zhaowenlan1779 commented on src/citra_qt/applets/swkbd.cpp:
```diff
@@ -104,7 +105,7 @@ void QtKeyboardDialog::HandleValidationError(Frontend::ValidationError error) {
         {ValidationError::EmptyInputNotAllowed, tr("Empty input is not allowed")},
     };
-    QMessageBox::critical(this, tr("Validation error"), VALIDATION_ERROR_MESSAGES.at(error));
+    MessageBox3D::critical(this, tr("Validation error"), VALIDATION_ERROR_MESSAGES.at(error));
```
Now that I think about it, will this work properly? It likely is trying to put two message boxes on both halves of the *keyboard dialog*, I think.

Plus, the keyboard dialog should be moved to the left eye.

zhaowenlan1779 commented on src/citra_qt/camera/qt_camera_base.cpp:
```diff
@@ -48,7 +49,7 @@ std::unique_ptr<CameraInterface> QtCameraFactory::CreatePreview(const std::strin
         return camera;
     }
-    QMessageBox::critical(
+    MessageBox3D::critical(
```
Won't this break? The `parent` is null here.

zhaowenlan1779 commented on src/citra_qt/compatdb.cpp:
```diff
@@ -68,8 +69,8 @@ void CompatDB::Submit() {
     if (!testcase_watcher.result()) {
-        QMessageBox::critical(this, tr("Communication error"),
-                              tr("An error occured while sending the Testcase"));
+        MessageBox3D::critical(this, tr("Communication error"),
```
Ditto. Won't this put two message boxes on the two halves of *the compatdb dialog*? 
Also, should move the compatdb dialog to left eye.

zhaowenlan1779 commented on src/citra_qt/configuration/configure_graphics.cpp:
```diff
@@ -32,7 +33,7 @@ ConfigureGraphics::ConfigureGraphics(QWidget* parent)
     connect(ui->toggle_hw_shader, &QCheckBox::stateChanged, this, [this](int state) {
         if (state == Qt::Checked) {
-            QMessageBox::warning(
+            MessageBox3D::warning(
```
ditto regarding the parent object, and this also applies to many occurrences below.

zhaowenlan1779 commented on src/citra_qt/main.cpp:
```diff
@@ -1675,8 +1680,8 @@ bool GMainWindow::ConfirmClose() {
-        QMessageBox::question(this, tr("Citra"), tr("Would you like to exit now?"),
-                              QMessageBox::Yes | QMessageBox::No, QMessageBox::No);
+        MessageBox3D::question(this, tr("Citra"), tr("Would you like to exit now?"),
+                               QMessageBox::Yes | QMessageBox::No, QMessageBox::No);
```
the constants here are not changed.

zhaowenlan1779 commented on src/citra_qt/util/sbs_dialog.h:
```diff
@@ -0,0 +1,35 @@
+/// move dialog to centre of left eye in SBS 3D mode
+void MoveDialogToLeftEye(QWidget* dialog, QWidget* parent, int offset = 0);
+
+// move dialog to centre of right eye in SBS 3D mode
```
should be `///`

zhaowenlan1779 commented on src/citra_qt/util/sbs_dialog.cpp:
```diff
@@ -0,0 +1,78 @@
+    QMessageBox::StandardButtons buttons, QMessageBox::StandardButton defaultButton) {
+    constexpr int parallax = 8;
+    QMessageBox left_box(icon, title, text, buttons, parent);
+    QMessageBox right_box(icon, title, text, buttons, parent);
```
this will create two message boxes whether 3D is enabled or not, I don't think this is what we want.
Also, might as well consider only enabling 3D message boxes when 3D is enabled *and game is running*.

B3n30 commented:
needs rebase and has unaddressed comments

jroweboy commented:
@zhaowenlan1779 addressed review comments. for the few cases where I didn't have a good idea of how to handle the 3D box, I just replaced it with QMessageBox which should be good enough for now imo.

zhaowenlan1779 commented:
This doesn't seem to be working. From my testing the right box only appears for a moment and then disappears

Dragios commented:
What happen to this PR? Almost approaching 1 year anniversary.
